File: Agregacion/E-9/Java-Eclipse/src/agregacion/Producto.java
package agregacion;
import java.util.ArrayList;
import java.util.List;

public class Producto {
	private String nombre;
    private double precio;

    // Constructor
    public Producto(String nombre, double precio) {
        this.nombre = nombre;
        this.precio = precio;
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    // Método para mostrar información
    public void mostrarInfo() {
        System.out.println("Producto: " + nombre + " - Precio: $" + precio);
    }
}
class CarritoCompras {
    private List<Producto> productos;
    private static final int LIMITE = 10;

    // Constructor
    public CarritoCompras() {
        this.productos = new ArrayList<>();
    }

    // Método para agregar producto
    public void agregarProducto(Producto producto) {
        if (productos.size() < LIMITE) {
            productos.add(producto);
            System.out.println("Producto '" + producto.getNombre() + "' agregado al carrito.");
        } else {
            System.out.println("No se puede agregar '" + producto.getNombre() + "'. El carrito está lleno (máximo " + LIMITE + " productos).");
        }
    }

    // Método para mostrar el carrito
    public void mostrarCarrito() {
        if (productos.isEmpty()) {
            System.out.println("El carrito está vacío.");
        } else {
            System.out.println("\nContenido del carrito (" + productos.size() + " productos):");
            System.out.println("--------------------------");
            for (Producto producto : productos) {
                producto.mostrarInfo();
            }
            System.out.println("--------------------------");
        }
    }

    // Getters (no se incluyen setters para la lista por seguridad)
    public List<Producto> getProductos() {
        return new ArrayList<>(productos); // Devuelve una copia para proteger la encapsulación
    }
}

File: Agregacion/E-9/Python/Producto.py
class Producto:
    def __init__(self, nombre, precio):
        self.__nombre = nombre
        self.__precio = precio
    
    @property
    def nombre(self):
        return self.__nombre
    
    @nombre.setter
    def nombre(self, nombre):
        self.__nombre = nombre
    
    @property
    def precio(self):
        return self.__precio
    
    @precio.setter
    def precio(self, precio):
        self.__precio = precio
    
    
    def mostrar_info(self):
        print(f"Producto: {self.__nombre} - Precio: {self.__precio:.2f} Bs")


class CarritoCompras:
    A = 10
    
    def __init__(self):
        self.__productos = []
    
    
    def agregar_producto(self, producto):
        if len(self.__productos) < self.A:
            self.__productos.append(producto)
            print(f"Producto '{producto.nombre}' agregado al carrito.")
        else:
            print(f"No se puede agregar '{producto.nombre}'. El carrito está lleno (máximo {self.A} productos).")
    
    def mostrar_carrito(self):
        if not self.__productos:
            print("El carrito está vacío.")
        else:
            print(f"\nContenido del carrito ({len(self.__productos)} productos):")
            print("--------------------------")
            for producto in self.__productos:
                producto.mostrar_info()
            print("--------------------------")
    
    @property
    def productos(self):
        return self.__productos.copy()



p1 = Producto("Whisky", 1200.99)
p2 = Producto("Sarten", 599.50)
p3 = Producto("Lata de Leche", 350.75)
p4 = Producto("Galletas", 89.99)


car = CarritoCompras()


car.agregar_producto(p1)
car.agregar_producto(p2)
car.agregar_producto(p3)
car.agregar_producto(p4)


car.mostrar_carrito()


for k in range(10):
    p = Producto(f"Producto extra {k+1}", 10.99)
    car.agregar_producto(p)

car.mostrar_carrito()